examples/decoding/parallel_decoding.py

```diff
@@ -0,0 +1,245 @@
+pool manager.
+"""
+
+# %%
+# Let's first define some utility functions for benchmarking and data
+# processing.  We'll also download a video and create a longer version by
+# repeating it multiple times. This simulates working with long videos that
+# require efficient processing. You can ignore that part and jump right below to
+# :ref:`start_parallel_decoding`.
+
+from typing import List
+import torch
+import requests
+import tempfile
+from pathlib import Path
+import subprocess
+from time import perf_counter_ns
+from datetime import timedelta
+
+from joblib import Parallel, delayed, cpu_count
+from torchcodec.decoders import VideoDecoder
+
+
+def bench(f, *args, num_exp=3, warmup=1, **kwargs):
+    """Benchmark a function by running it multiple times and measuring execution time."""
+    for _ in range(warmup):
+        f(*args, **kwargs)
+
+    times = []
+    for _ in range(num_exp):
+        start = perf_counter_ns()
+        result = f(*args, **kwargs)
+        end = perf_counter_ns()
+        times.append(end - start)
+
+    return torch.tensor(times).float(), result
+
+
+def report_stats(times, unit="s"):
+    """Report median and standard deviation of benchmark times."""
+    mul = {
+        "ns": 1,
+        "µs": 1e-3,
+        "ms": 1e-6,
+        "s": 1e-9,
+    }[unit]
+    times = times * mul
+    std = times.std().item()
+    med = times.median().item()
+    print(f"median = {med:.2f}{unit} ± {std:.2f}")
+    return med
+
+
+def split_indices(indices: List[int], num_chunks: int) -> List[List[int]]:
+    """Split a list of indices into approximately equal chunks."""
+    chunk_size = len(indices) // num_chunks
+    chunks = []
+
+    for i in range(num_chunks - 1):
+        chunks.append(indices[i * chunk_size:(i + 1) * chunk_size])
+
+    # Last chunk may be slightly larger
+    chunks.append(indices[(num_chunks - 1) * chunk_size:])
+    return chunks
+
+
+# Video source: https://www.pexels.com/video/dog-eating-854132/
+# License: CC0. Author: Coverr.
+url = "https://videos.pexels.com/video-files/854132/854132-sd_640_360_25fps.mp4"
+response = requests.get(url, headers={"User-Agent": ""})
+if response.status_code != 200:
+    raise RuntimeError(f"Failed to download video. {response.status_code = }.")
+
+temp_dir = tempfile.mkdtemp()
+short_video_path = Path(temp_dir) / "short_video.mp4"
+with open(short_video_path, 'wb') as f:
+    for chunk in response.iter_content():
+        f.write(chunk)
+
+# Create a longer video by repeating the short one 50 times
+long_video_path = Path(temp_dir) / "long_video.mp4"
+ffmpeg_command = [
+    "ffmpeg", "-y",
+    "-stream_loop", "49",  # repeat video 50 times
+    "-i", str(short_video_path),
+    "-c", "copy",
+    str(long_video_path)
+]
+subprocess.run(ffmpeg_command, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+decoder = VideoDecoder(long_video_path, seek_mode="approximate")
+metadata = decoder.metadata
+
```

Comment:
Can we put most of the logic above that appears at module scope into a function, and then call that? It's a big chunk of code, and it's not obvious what from it lives on. I think having the stuff above it helps readers understand what's going on. So I think something like this:
```python
def generate_long_video():
    # most of this logic

short_duration_path, long_duration_path = generate_long_video()
```